ui/src/app/views/rename-modal/component.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+
+import {
+  Button
+} from 'app/components/buttons';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+} from 'app/components/modals';
+import {ReactComponent} from 'app/utils';
+
+
+interface RenameModalProps {
+  resource: any;
+  resourceType: string;
+  onRename: Function;
+  onCancel: Function;
+}
+interface RenameModalState {
+  loading: boolean;
+  newName: string;
+}
+
+@ReactComponent({
+  selector: 'app-rename-modal',
+  propNames: ['resource', 'resourceType', 'onRename', 'onCancel']
+})
+export class RenameModalComponent extends React.Component<RenameModalProps, RenameModalState> {
+  constructor(props: RenameModalProps) {
+    super(props);
+    this.state = {
+      loading: false,
+      newName: ''
+    };
+  }
+
+  private emitRename(): void {
+    if (this.state.loading) {
+      return;
+    }
+    this.setState({
+      loading: true
+    });
+    this.props.onRename({
+      name: this.props.resource.name,
+      newName: this.state.newName
+    });
+  }
+
+  render() {
+    return <React.Fragment>
+      <Modal>
+        <ModalTitle>Please enter the new name for {this.props.resource.name}</ModalTitle>
```

Comment:
@panentheos A lot of React code I've seen pulls props out to local constants in the render() method. Do you recommend this pattern and would you apply it here (i.e. for the resource name)? (I assume it's purely for code readability).